crates/aptos-rest-client/src/faucet.rs

```diff
@@ -31,61 +41,62 @@ impl FaucetClient {
         let mut url = self.faucet_url.clone();
         url.set_path("mint");
         let query = format!("auth_key={}&amount=0&return_txns=true", address);
         url.set_query(Some(&query));
 
-        let response = client.post(url).send().map_err(Error::request)?;
+        let response = self.inner.post(url).send().await.map_err(Error::request)?;
         let status_code = response.status();
-        let body = response.text().map_err(Error::decode)?;
+        let body = response.text().await.map_err(Error::decode)?;
         if !status_code.is_success() {
             return Err(anyhow::anyhow!("body: {}", body));
         }
 
         let bytes = hex::decode(body).map_err(Error::decode)?;
         let txns: Vec<SignedTransaction> = bcs::from_bytes(&bytes).map_err(Error::decode)?;
 
-        tokio::runtime::Runtime::new()
-            .unwrap()
-            .block_on(self.rest_client.wait_for_signed_transaction(&txns[0]))
+        self.rest_client
+            .wait_for_signed_transaction(&txns[0])
+            .await
             .map_err(Error::unknown)?;
 
         Ok(())
     }
 
-    pub fn fund(&self, address: AccountAddress, amount: u64) -> Result<()> {
-        let client = reqwest::blocking::Client::new();
+    /// Fund an account with the given amount.
+    pub async fn fund(&self, address: AccountAddress, amount: u64) -> Result<()> {
         let mut url = self.faucet_url.clone();
         url.set_path("mint");
         let query = format!("auth_key={}&amount={}&return_txns=true", address, amount);
         url.set_query(Some(&query));
 
         // Faucet returns the transaction that creates the account and needs to be waited on before
         // returning.
-        let response = client.post(url).send().map_err(Error::request)?;
+        let response = self.inner.post(url).send().await.map_err(Error::request)?;
         let status_code = response.status();
-        let body = response.text().map_err(Error::decode)?;
+        let body = response.text().await.map_err(Error::decode)?;
         if !status_code.is_success() {
             return Err(Error::status(status_code.as_u16()).into());
         }
 
         let bytes = hex::decode(body).map_err(Error::decode)?;
         let txns: Vec<SignedTransaction> = bcs::from_bytes(&bytes).map_err(Error::decode)?;
 
-        tokio::runtime::Runtime::new()
-            .unwrap()
-            .block_on(self.rest_client.wait_for_signed_transaction(&txns[0]))
+        self.rest_client
+            .wait_for_signed_transaction(&txns[0])
+            .await
             .map_err(Error::unknown)?;
 
         Ok(())
     }
 
-    pub fn mint(&self, address: AccountAddress, amount: u64) -> Result<()> {
-        self.create_account(address)?;
-        self.fund(address, amount)?;
+    // Create and fund an account.
+    pub async fn mint(&self, address: AccountAddress, amount: u64) -> Result<()> {
+        self.create_account(address).await?;
+        self.fund(address, amount).await?;
```

Comment:
we should stop using this methodology and just use account::transfer --- or the function account_transfer.